### PR TITLE
feat(template): Make rxIf stable

### PR DESCRIPTION
**Pull request by hoebbelsB**
# Description

* [x] adjust `rxIf` API to be in line with ngIf
* [x] implemented rxIf test-suite based on original [ngIfs test-suite](https://github.com/angular/angular/blob/main/packages/common/test/directives/ng_if_spec.ts)
* [x] adjust rxIf template handling in order to properly display values
* [x] find a solution for not implemented ability to switch templates dynamically
  * `should support removing the then/else templates`
  * `should support dynamic else`
* [x] implement test cases for suspense, complete, error
* [x] implement test cases for observable and strategy handling
* [x] create JS docs for `rxIf` 
* [x] move `rxIf` to `@rx-angular/template/if`

### CODE REVIEW

**Review by edbzn — APPROVED** [review]

File: libs/template/experimental/if/src/lib/if.directive.ts [hunk 1]
```diff
@@ -48,21 +48,20 @@ export class RxIf<U> implements OnInit, OnDestroy, OnChanges {
     rxIfTemplateNames
   >;
 
-  @Input()
-  set rxIf(potentialObservable: Observable<U> | U | null | undefined) {
-    this.observablesHandler.next(coerceObservable(potentialObservable));
-  }
+  private lastValidValue: U | null;
```
Comment (edbzn) [inline libs/template/experimental/if/src/lib/if.directive.ts:51]: General thought, what about using the private instance fields syntax? So we make sure these fields are not accessible at runtime.

```suggestion
  #lastValidValue: U | null;
```

File: libs/template/experimental/if/src/lib/tests/if.directive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { RX_RENDER_STRATEGIES_CONFIG } from '@rx-angular/cdk/render-strategies';
+import { IfModule } from '../if.module';
+import { createTestComponent, TestComponent } from './fixtures';
+
+{
```
Comment (edbzn) [inline libs/template/experimental/if/src/lib/tests/if.directive.spec.ts:7]: These braces are superfluous.

File: libs/template/experimental/if/src/lib/tests/if.directive.observable.spec.ts [hunk 3]
```diff
@@ -0,0 +1,483 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { RX_RENDER_STRATEGIES_CONFIG } from '@rx-angular/cdk/render-strategies';
+import { BehaviorSubject, of, startWith, throwError, tap } from 'rxjs';
+import { IfModule } from '../if.module';
+import { createTestComponent, TestComponent } from './fixtures';
+
+{
```
Comment (edbzn) [inline libs/template/experimental/if/src/lib/tests/if.directive.observable.spec.ts:8]: These braces are superfluous.

**Review by edbzn — APPROVED** [review]

**Comment by edbzn** [comment]
Question, does it fix the following issue https://github.com/rx-angular/rx-angular/issues/1335?

File: libs/template/experimental/if/src/lib/if.directive.ts (hunk 1 above)
Comment (hoebbelsB) [reply libs/template/experimental/if/src/lib/if.directive.ts:51]: If we go that route I think we should do it in a separate PR for all of our directives/components

**Review by Karnaukhov-kh — APPROVED** [review]

**Comment by edbzn** [comment]
Tested in my todo mvc app, it's working well!

**Pull request merged**